doltpy/core/write/write.py

```diff
@@ -146,41 +149,33 @@ def import_dict(repo: Dolt,
     assert row_count > 0, 'Must provide at least a single row'
     assert all(len(val_list) == row_count for val_list in data.values()), 'Must provide value lists of uniform length'
 
+    # Get an Engine object
+    engine = repo.get_engine(host=dolt_server_host, port=dolt_server_port)
+
     # If the table does not exist, create it using type inference to build a create statement
     if import_mode == CREATE:
         assert primary_keys, 'primary_keys need to be provided when inferring a schema'
-        _create_table_inferred(repo, table_name, data, primary_keys, dolt_server_host, dolt_server_port)
+        _create_table_inferred(engine, table_name, data, primary_keys)
 
     # Now transform the data to lists of tuples, where the elements are in the same order as the
     # columns when sorted lexicographically
-    sorted_cols = sorted(data.keys())
-    tuple_list = []
+    # sorted_cols = sorted(data.keys())
+    rows = []
     for i in range(row_count):
-        tuple_list.append(tuple([data[col][i] for col in sorted_cols]))
+        rows.append({col: data[col][i] for col in data.keys()})
 
-    insert_statement = _get_insert_statement(table_name, sorted_cols)
     logger.info('Inserting {row_count} rows into table {table_name}'.format(row_count=row_count,
                                                                             table_name=table_name))
+    table = MetaData(bind=engine, reflect=True).tables[table_name]
+    for i in range(max(1, math.ceil(len(rows) / chunk_size))):
+        chunk_start, chunk_end = i * chunk_size, min((i+1) * chunk_size, len(rows))
```

Comment:
we should call this `batch_size`, "chunks" mean something else in Dolt
also, nit: this multiple assignment is too dense